flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -600,15 +612,24 @@ public void restoreStateHandles(
                                 knownPhysicalFiles.computeIfAbsent(
                                         fileHandle.getFilePath(),
                                         path -> {
+                                            boolean managedByFileMergingManager =
+                                                    isManagedByFileMergingManager(
+                                                            path,
+                                                            subtaskKey,
+                                                            fileHandle.getScope());
                                             PhysicalFileDeleter fileDeleter =
-                                                    (isManagedByFileMergingManager(
-                                                                    path,
-                                                                    subtaskKey,
-                                                                    fileHandle.getScope()))
+                                                    managedByFileMergingManager
```

Comment:
Given that there is `owned` attribute within the `PhysicalFile`, how about remove this diversion and wrap everything within the `PhysicalFile`?

Reply:
I moved the setting of `FileDeleter` into `PhysicalFile`, and left `SpaceStat.update()` in `FileMergingSnapshotManager`